pkg/controller/control.go

```diff
@@ -1235,6 +1244,29 @@ func (c *Controller) handleErrorNoLock(err error) error {
 	return err
 }
 
+func (c *Controller) handleDiskNoSpaceErrorForReplicas(replicaNoSpaceErrMap map[string]string) error {
+	replicaWritableCount := 0
+	for _, r := range c.replicas {
+		if r.Mode == types.ERR {
+			continue
+		}
+		replicaWritableCount++
+	}
+
+	// https://github.com/longhorn/longhorn/issues/10718#issuecomment-3190666485
+	// If a replica is on a disk that is out of space, and others are not, this replica will be marked as ERR to avoid the inconsistent data.
+	// The volume will be readable when all disks of the replicas are out of space so last one replica will not be marked as ERR because of ENOSPC.
+	if len(replicaNoSpaceErrMap) != 0 {
```

Comment:
Suggest returning early when `len(replicaNoSpaceErrMap)` is empty.

Reply:
Modified. Thanks.